src/Bindings/LuaState.h

```diff
@@ -112,6 +115,63 @@ class cLuaState
 	} ;
 	
 	
+	/** Represents a callback to Lua that C++ code can call.
+	Is thread-safe and unload-safe.
+	When the Lua state is unloaded, the callback returns an error instead of calling into non-existent code.
+	To receive the callback instance from the Lua side, use RefStack() or (better) cLuaState::GetStackValue(). */
+	class cCallback
+	{
+	public:
+		/** Creates an unbound callback instance. */
+		cCallback(void) = default;
+
+		~cCallback()
+		{
+			Clear();
+		}
+
+		/** Calls the Lua callback, if still available.
+		Returns true if callback has been called.
+		Returns false if the Lua state isn't valid anymore. */
+		template <typename... Args>
+		bool Call(Args &&... args)
```

Comment:
Yes, and that's why I wrote a little experiment to help me understand that better: https://github.com/madmaxoft/LambdaLifetimeTest

I've found that the lambdas do stay valid, because the code uses excessive copying. Two things that are somewhat troubling is that the lambda's by-reference captures go out of scope, yet the lambda can access them (-> possible crash with accessing invalid memory) and we can't pass in non-copyable objects (such as Lua `cCallback`)